### PR TITLE
promisify and lipstickify

### PR DESCRIPTION
**Pull request by SamVerschueren**
Did quite some changes/improvements so might consider a good review before merging :).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,58 +1,48 @@
...
+const cheerio = require('cheerio');
+const uncss = require('uncss');
+const pify = require('pify');
+
+const uncssP = pify(uncss);
 
-		var ret = [];
-		var $ = cheerio.load(data);
+const getCSS = () => {
+	return got('https://github.com').then(response => {
```
Comment (sindresorhus) [inline index.js]: You can just pass `got` directly, no need for `return`.

File: index.js [hunk 2]
```diff
@@ -122,38 +102,10 @@ function cleanupCss(str) {
...
 };
+
+module.exports = () => getRenderedFixture()
+	.then(html => getCSS()
+		.then(stylesheets => uncssP(html, {stylesheets, ignore: [/^\.pl/]})))
+	.then(css => cleanupCss(css));
```
Comment (sindresorhus) [inline index.js]:
```js
.then(cleanupCss);
```

File: index.js [hunk 3]
```diff
@@ -122,38 +102,10 @@ function cleanupCss(str) {
...
 };
+
+module.exports = () => getRenderedFixture()
+	.then(html => getCSS()
+		.then(stylesheets => uncssP(html, {stylesheets, ignore: [/^\.pl/]})))
```
Comment (sindresorhus) [inline index.js]: I would use Promise.all and get the HTML and CSS at the same time.

File: test.js [hunk 4]
```diff
@@ -1,13 +1,7 @@
...
-	this.timeout(500000);
-
-	githubMarkdownCss(function (err, css) {
-		assert(!err, err);
-		assert(/markdown-body/.test(css));
-		cb();
-	});
+test(async t => {
+	const result = await m();
+	t.true(/markdown-body/.test(result));
```
Comment (sindresorhus) [inline test.js]: `t.regex`

Comment (sindresorhus) [inline test.js]: And just put the `await m()` inline.

**Comment by sindresorhus** [comment]
Nice! Looks so much simpler now ❤️

File: index.js [hunk 5]
```diff
@@ -1,58 +1,42 @@
...
-
-		var $ = cheerio.load(data);
-		var html = $('.markdown-body').parent().html();
+	return ret;
+});
 
-		cb(null, html);
-	});
-}
+const getRenderedFixture = () => got('https://github.com/sindresorhus/generate-github-markdown-css/blob/master/fixture.md').then(response => {
```
Comment (SamVerschueren) [inline index.js]: Is it ok to put the url inline here?

Comment (sindresorhus) [inline index.js]: No, make it a top-level `const` variable above here.

**Comment by SamVerschueren** [comment]
done

**Comment by sindresorhus** [comment]
Thanks Sam :)

happy

**Comment by SamVerschueren** [comment]
You're more then welcome :)

**Pull request merged**